### ISSUE TITLE
[Feature request]: Allow runtimes setting permissions

### ISSUE THREAD

**Issue by Bleuzen** [flatpak/flatpak#4633]
### Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/flatpak/flatpak/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/flatpak/flatpak/issues) for a feature request that matches the one I want to file, without success.

### Suggestion

Currently there are many apps requiring the exact same permissions.
For example:

To make theming work in case of custom css of gtk themes (prominent example: KDE Plasmas default dark theme: Breeze Dark) all gtk applications need read access to `.config/gtk-3.0`.
The current solution would be to add
```
-filesystem=xdg-config/gtk-3.0:ro
```
to any gtk app. This could be simplified by this feature requested here. If we could add this as a default permission for the freedesktop (or gtk) runtimes, this filesystem permission would not have to be added hundreds of times in all gtk apps and make theming work out of the box.

**Comment by TomaszGasior** [comment]
This can be done downstream by distribution who wants to suport  that KDE Plasma feature. Just provide `/var/lib/flatpak/overrides/global` with this content:

```
[Context]
filesystems=xdg-config/gtk-3.0;xdg-config/gtk-4.0;
```

**Comment by Bleuzen** [comment]
> This can be done downstream by distribution which wants to suport that KDE Plasma feature. Just provide `/var/lib/flatpak/overrides/global`

This is a known workaround for some time now. But please be realistic: Downstream distros won't fix it. This issue exists for long enough and no distribution ships with this workaround. Most don't even know because it is impossible for every distro developer to test everything.
So we have these options:

A) Fix it in hundred places and ask all of distro developers to ship this workaround with their package/distro. Many will forget or don't care. What if something changes about this? Would have to ask many people again to fix it another time in hundreds of places.

B) Fix it in only one place (flatpak/runtimes). Makes it work for everyone ootb. Users happy, distro maintainer happy.

Think B makes much more sense.

**Comment by TomaszGasior** [comment]
No, with B solution you are creating security hole. Just for customization. Customization is not core feature of the OS. If you are able to create gtk.css or install addons which manage gtk.css customization file, you are also able (or that tools are) to add needed global override to flatpak.

**Comment by Bleuzen** [comment]
> No, with B solution you are creating security hole

Checked that directory and can only find the ini, css files and svgs. Can't imagine what apps can do evil with readonly css/svg files.
In which case is there sensible data in `.config/gtk-3.0/`?

**Comment by TomaszGasior** [comment]
Yes, `servers` and `bookmarks` for example.

**Comment by Bleuzen** [comment]
> Yes, servers and bookmarks for example.

Hm we could allow only the files needed:
```
--filesystem=xdg-config/gtk-3.0/settings.ini:ro
--filesystem=xdg-config/gtk-3.0/gtk.css:ro
--filesystem=xdg-config/gtk-3.0/colors.css:ro
--filesystem=xdg-config/gtk-3.0/assets:ro
```

**Comment by Erick555** [comment]
KDE runtime is able to [set permissions](https://invent.kde.org/packaging/flatpak-kde-runtime/-/blob/qt5.15lts-21.08/org.kde.Sdk.json.in#L77) inherited by all apps which use it. Flatpak could mount those paths by default if developers wanted to but I'm afraid they don't.

**Comment by Bleuzen** [comment]
> KDE runtime is able to [set permissions](https://invent.kde.org/packaging/flatpak-kde-runtime/-/blob/qt5.15lts-21.08/org.kde.Sdk.json.in#L77) inherited by all apps which use it.

Thanks for pointing that out.

> Flatpak could mount those paths by default if developers wanted to but I'm afraid they don't.

If it doesn't have any downsides, but makes people happy, I can't see why they wouldn't want it.
Will see, I suggested it here:
https://gitlab.com/freedesktop-sdk/freedesktop-sdk/-/issues/1362
